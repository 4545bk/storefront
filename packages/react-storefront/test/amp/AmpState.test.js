/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import AmpState from '../../src/amp/AmpState'
import { Provider } from 'mobx-react'
import { mount } from 'enzyme'

describe('AmpState', () => {
  it('should render an amp-state when amp is true', () => {
    expect(
      mount(
        <Provider app={{ amp: true }}>
          <AmpState initialState={{ foo: 'bar' }}>
            <div>inner</div>
          </AmpState>
        </Provider>,
      ),
    ).toMatchSnapshot()
  })
  it('should render only children when amp is false', () => {
    expect(
      mount(
        <Provider app={{ amp: false }}>
          <AmpState>
            <div>inner</div>
          </AmpState>
        </Provider>,
      ),
    ).toMatchSnapshot()
  })
})
