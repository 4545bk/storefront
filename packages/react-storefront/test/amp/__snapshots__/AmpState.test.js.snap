// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmpState should render an amp-state when amp is true 1`] = `
<Provider app={{...}}>
  <inject-AmpState initialState={{...}}>
    <AmpState initialState={{...}} amp={true} id="moovAmpState">
      <Provider ampStateId="moovAmpState">
        <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
          <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
            <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
          </SideEffect(NullComponent)>
        </HelmetWrapper>
        <amp-state id="moovAmpState">
          <script type="application/json" dangerouslySetInnerHTML={{...}} />
        </amp-state>
        <div>
          inner
        </div>
      </Provider>
    </AmpState>
  </inject-AmpState>
</Provider>
`;

exports[`AmpState should render only children when amp is false 1`] = `
<Provider app={{...}}>
  <inject-AmpState>
    <AmpState amp={false} initialState={{...}} id="moovAmpState">
      <Provider ampStateId="moovAmpState">
        <div>
          inner
        </div>
      </Provider>
    </AmpState>
  </inject-AmpState>
</Provider>
`;
